packages/ERTP/src/paymentLedger.js

```diff
@@ -200,10 +200,6 @@ export const preparePaymentLedger = (
     }
   };
 
-  /** @type {(left: Amount, right: Amount) => Amount} */
```

Comment:
I presume that's because the switch from one representation to the other will be non-atomic. In that case, I think we need to be more explicit about the transition.
#8682 shows how it's possible to leave the old Kind around.

If we expect to transition by requiring existing contracts to explicitly move assets from one kind to another and discard the redundant one, it might make sense to have explicit support to create a new purse from an old one, rather than requiring every users to create an intermediate payment to do the work.

Reply:
a helper probably won't hurt, but "every user" is exactly walletFactory, unless I'm missing something. In the future, Zoe would always use the mutable store for NFT purses.

Reply:
> ... paymentLedger ... ?

My intention is only to make this representation change for purses, not for payments. Purses can accumulate large amounts of assets "at rest". Payments, allocations, and amounts in proposals are all about assets "in flight" and we hope will typically be small.

Reply:
What I mean is: we still need the current behavior for purses. We can introduce a new kind, but getting rid of the old one takes another few steps.

Reply:
> we still need the current behavior for purses. We can introduce a new kind, but getting rid of the old one takes another few steps.

If we introduce a new kind, we still have to keep defining the old one. AIUI, the exo system doesn't have a mechanism for deleting previously-defined kinds, and upgrade will fail if any existing Kind doesn't get assigned a behavior.

Reply:
I was thinking maybe if we got to the point that there are no objects of the old kind, we could get rid of the old kind. But if exo doesn't support that, read "another few steps" as including enhancing exo to be able to do that :)

Reply:
Kind as in assetKind? Something in addition to the existing `nat`, `set`, `copySet`, `copyBag`? That was not my plan. But if there's a good reason we could consider it. Why?

My current plan is that the difference is completely encapsulated within the purses in question.

Reply:
Introducing a new assetKind potentially touches a lot more code, and raises a lot more version slippage questions. Nevertheless, we did built the system so we could grow new assetKinds later if there's a good reason.

Reply:
> Kind as in assetKind? Something in addition to the existing `nat`, `set`, `copySet`, `copyBag`?

No. Kind as in behavior.

> A VDO has a "kind", which defines what sort of behavior and state it will possess. 
-- https://github.com/Agoric/agoric-sdk/blob/master/packages/SwingSet/docs/virtual-objects.md

Reply:
Ah, of course. Thanks. Makes a lot more sense ;)

Reply:
These points all become relevant to the future change that leverages this abstraction to make an actual change to a Purse's durable representation. But since this PR does not make any such change, resolving.